WalletWasabi/WabiSabi/Client/CoinJoinCoinSelectorRandomnessGenerator.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Linq;
+using WabiSabi.Crypto.Randomness;
+
+namespace WalletWasabi.WabiSabi.Client;
+
+/// <summary>
+/// Generator of randomness for <see cref="CoinJoinCoinSelector"/>.
+/// </summary>
+public class CoinJoinCoinSelectorRandomnessGenerator
+{
+	public CoinJoinCoinSelectorRandomnessGenerator(WasabiRandom rnd)
+	{
+		Rnd = rnd;
+	}
+
+	public WasabiRandom Rnd { get; }
+
+	public virtual int GetRandomBiasedSameTxAllowance(int percent)
+	{
+		for (int num = 0; num <= 100; num++)
+		{
+			if (Rnd.GetInt(1, 101) <= percent)
+			{
+				return num;
+			}
+		}
+
+		return 0;
+	}
+
+	/// <summary>
+	/// Calculates how many inputs are desirable to be registered.
+	/// Note: Random biasing is applied.
+	/// </summary>
+	/// <returns>A number: 1, 2, .., <see cref="CoinJoinCoinSelector.MaxInputsRegistrableByWallet"/>.</returns>
+	public virtual int GetInputTarget()
+	{
+		// Until our UTXO count target isn't reached, let's register as few coins as we can to reach it.
+		int targetInputCount = CoinJoinCoinSelector.MaxInputsRegistrableByWallet;
```

Comment:
It's a matter of opinion and not much important but I think it would be nicer if `MaxInputsRegistrableByWallet` was passed to `CoinJoinCoinSelectorRandomnessGenerator` in its constructor. This avoids the mutual dependency between `CoinJoinCoinSelectorRandomnessGenerator` and `CoinJoinCoinSelector`.

Reply:
Feel free to resolve this conversation.